File: firebase/functions/src/database/seat-finder/state-change/UserStateChange.ts
import {SeatFinderRequestType} from "../../../enum/SeatFinderRequestType";
import {UserStateType} from "../../../enum/UserStateType";
import {UserStateChangeReason} from "../../../enum/UserStateChangeReason";

export interface UserStateChange {
    requestType: SeatFinderRequestType;
    timestamp: number;
    resultState: UserStateType;
    reason: UserStateChangeReason;
    success: boolean;
}

File: core/model/src/main/java/io/github/jeddchoi/model/UserStateChange.kt
package io.github.jeddchoi.model

import kotlinx.datetime.Instant

data class UserStateChange(
    val requestType: SeatFinderUserRequestType = SeatFinderUserRequestType.Quit,
    val success: Boolean = false,
    val resultSeatState: SeatStateType = SeatStateType.Empty,
    val resultUserState: UserStateType = UserStateType.None,
    val reason: UserStateChangeReason = UserStateChangeReason.UserAction,
    val timestamp: Instant = Instant.DISTANT_FUTURE,
)
